### PR TITLE
Post-feedback from initial interview round (part 2)

### PR DESCRIPTION
**Pull request by nealcheng-guideline**
## Changes

1. Tell the candidate that we will import utils.py import to_decimal
2. Make a validation for make_query to ensure the prompt is being filled in. 
3. Be more specific about serialize in construct_prompt by mentioning dumping to json. 
4. Break apart prompt formatting into "participant -> json" and "string formatting"

### CODE REVIEW

File: problem/question2/validation.py [hunk 1]
```diff
@@ -17,4 +17,26 @@ def validate_model_input(api_key: str, model:str, temperature: float):
 
 def validate_prompt_input(participants: list[Participant]):
     if not all(isinstance(participant, Participant) for participant in participants):
-        raise TypeError("The argument for construct_prompt has to be a list of instance <Participants>")
\ No newline at end of file
+        raise TypeError("The argument for construct_prompt has to be a list of instance <Participants>")
+  
+    
+def validate_query(prompt: str):
```
Comment (ykim-gdl) [inline problem/question2/validation.py:23]: Q: I thought we wanted to validate whether the participant instances are correctly serialized to json?

Comment (nealcheng-guideline) [reply problem/question2/validation.py:23]: I've realized it's difficult because the participants will be a list and not a true json.  I would have to create some sort of a delimiting marker between the template and the participants so that I can extract the json.  Then convert to a dictionary.  

Let me look into this.

Comment (nealcheng-guideline) [reply problem/question2/validation.py:23]: The argument for not need to do this is:  if they can't dump it into the prompt in the first place, they would get additional errors beforehand.  If they CAN, then it should be correct.

Comment (nealcheng-guideline) [reply problem/question2/validation.py:23]: I believe the Decimal class isn't serialized via text.  Below is the prompt:
```
Prompt:

You are a financial accountant with 20 years of experience working within large firms
    such as Deloitte and PwC.  Can you please construct a recommendation for suitable 401k options for the employees within
    this legal entity?
    (Please note that participants will be denoted by user_id, which is their unique identifier):
     [{'user_id': 202, 'full_name': '', 'salary_dollars': '4000'}, {'user_id': 205, 'full_name': '', 'salary_dollars': '0'}] 
```

Comment (nealcheng-guideline) [reply problem/question2/validation.py:23]: My concern about making the test TOO narrow is that we could potentially be missing edge cases that could cause the candidate to not pass.

Comment (ykim-gdl) [reply problem/question2/validation.py:23]: if we're not confident about validating the serialization correctly, im fine with not checking that - do we still want to keep this new function `validate_query` in this case?

Comment (nealcheng-guideline) [reply problem/question2/validation.py:23]: Oh I see.  That's a good one, but my question is whether we want to fail or cause a candidate to be stuck if they've passed `participants` as is (as with John's candidate).  I think using `participants` as is will at least show that they know how to perform string formatting.  

I would probably follow up with the question **Can you take a look at the prompt to see if providing class names to the LLM makes sense?**

Comment (nealcheng-guideline) [reply problem/question2/validation.py:23]: I can roll back the `validate_query` commit

Comment (nealcheng-guideline) [reply problem/question2/validation.py]: Reverted in [Revert "prompt validation"](https://github.com/nealcheng-guideline/ds-team-coding-interview/pull/10/commits/6dc1634daba35a128c7d9321c15907c1efa8f524)

File: solution/question2/main.py [hunk 2]
```diff
@@ -27,11 +27,15 @@ def construct_prompt(participants: list[Participant]) -> str:
...
+    # TODO: Serialize the Participant objects to JSON
+    participant_info = [participant.model_dump() for participant in participants]
+    
+    # TODO: Add this information to the prompt
     prompt = f"""You are a financial accountant with 20 years of experience working within large firms
     such as Deloitte and PwC.  Can you please construct a recommendation for suitable 401k options for the employees within
     this legal entity?
     (Please note that participants will be denoted by user_id, which is their unique identifier):
-     {[participant.model_dump() for participant in participants]} """
+     {participant_info} """
```
Comment (ykim-gdl) [inline solution/question2/main.py:38]: Just a note: seems like this multi-line f-string works here but somehow this is not working in coderbyte..

Comment (nealcheng-guideline) [reply solution/question2/main.py:38]: Yeah just tested and agreed that it appears to be a platform issue.
